client/rpcserver/rpcserver.go

```diff
@@ -258,14 +252,19 @@ func (s *RPCServer) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 		}
 	}()
 
+	// Configure the websocket handler before starting the server.
+	s.mux.Get("/ws", func(w http.ResponseWriter, r *http.Request) {
+		s.wsServer.HandleConnect(ctx, w, r)
```

Comment:
Only way to inject the context into the websocket client since the r.Context() gets canceled the moment ServerHTTP returns.